ctapipe/io/hessiofilereader.py

```diff
@@ -0,0 +1,211 @@
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+from ctapipe.core import Provenance
+from ctapipe.io.eventfilereader import EventFileReader
+from ctapipe.io.containers import DataContainer
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription
+
+
+class HessioFileReader(EventFileReader):
+    """
+    EventFileReader for the hessio data format.
+
+    This class utilises `pyhessio` to read the hessio file, and stores the
+    information into the event containers.
+    """
+    _count = 0
+
+    def __init__(self, config, tool, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        try:
+            from pyhessio import open_hessio
+            from pyhessio import close_file
+            from pyhessio import HessioGeneralError
+        except ImportError:
+            msg = "The `pyhessio` python module is required to access MC data"
+            self.log.error(msg)
+            raise
+
+        self.open_hessio = open_hessio
+        self.close_hessio = close_file
+        self.HessioGeneralError = HessioGeneralError
+
+        self.allowed_tels = None
+
+        if HessioFileReader._count > 0:
+            self.log.warn("Only one pyhessio reader allowed at a time. "
+                          "Previous hessio file will be closed.")
+            self.close_hessio()
+        HessioFileReader._count += 1
+
+        self._metadata['is_simulation'] = True
+
+    @staticmethod
+    def is_compatible(file_path):
+        return file_path.endswith('.gz')
```

Comment:
What if other readers also read gzipped files, like "something.fits.gz" ? Then the factory **wrongly** tries to use this reader, if this reader happens to be asked first ..

Reply:
If that situation arises then we need to improve this function to better detect what a compatible file it. But we will do that when the situation occurs.

Reply:
Sounds fair enough

Reply:
yeah, in the future we can use a module like *magic* that detects file types by magic numbers or mime types. Depends on how many variants we end up with! At least here, you might want to check for `".simtel.gz"`

Reply:
EventIO file are quite easy to identifiy, they all start with two special magic bytes. gz files as well.

So we could do 
```python
with open(path, 'rb') as f:
    marker_bytes = f.read(4)
if  marker_bytes[0] == 0x1f and marker_bytes[1] == 0x8b:
    with gzip.open(path, 'rb') as f:
        marker_bytes = f.read(4)

return struct.unpack('I', marker_bytes)[0] == 3558836791
```

Reply:
First case is little endian, second case big endian

Reply:
Sounds good, I'll add that now